src/core/portablemc/__init__.py

```diff
@@ -775,6 +790,19 @@ def cleanup():
                             with open(path.join(bin_dir, native_name), "wb") as native_file:
                                 shutil.copyfileobj(native_zip_file, native_file)
 
+        for bin_file in self.bin_files:
+            bin_name = path.basename(bin_file)
+            # Here we try to remove the version numbers of .so files.
+            so_idx = bin_name.rfind(".so")
+            if so_idx >= 0:
+                bin_name = bin_name[:so_idx + len(".so")]
+            # Try to symlink the file in the bin dir, and fallback to simple copy.
+            bin_dst_file = path.join(bin_dir, bin_name)
+            try:
+                os.symlink(bin_file, bin_dst_file)
```

Comment:
Before even this is attempted it should be ensured that `bin_file` even exists on the filesystem and perhaps throw an error otherwise?